### PR TITLE
Docs: Updates for Storybook Test

### LINKED ISSUE #29794: Document Component Testing changes in 8.5
# Focused Tests

We can expand the section in https://storybook.js.org/docs/writing-tests/test-addon#storybook-ui
No code snippets necessary, it's all explainable with prose+video/images, similar to the existing content in that section.

Focused tests work on the story, component and group level.

Coverage is not enabled in focused tests. You _can_ run a focused test while having watch mode enabled simultanously.

# Coverage

- Coverage only accounts for _stories_, not any other Vitest tests configured. Using Vitest CLI instead includes the full coverage of the whole Vitest test suite, including stories.
- We report _Statements_ as the overall project coverage metric
- We respect `test.coverage.watermarks` as a config for the pie chart color
- You need the Vitest dependencies
- Coverage and watch-mode are mutually exclusive. If coverage is enabled, enabling watch mode disables coverage.
- HTML report is available at `localhost:6006/coverage/index.html`.
- We respect most of the the user's settings at `test.coverage` (providers, etc), except for:
```
            enabled: true,
            clean: false,
            cleanOnRerun: !watchMode,
            reportOnFailure: true,
            reporter: [['html', {}], storybookCoverageReporter],
            reportsDirectory: resolvePathInStorybookCache(COVERAGE_DIRECTORY),
```

# Misc

### PR DESCRIPTION
**Pull request by kylegach**
Closes #29794

## What I did

- Focused tests
- Coverage
- A11y
- Remove `test.include` & `test.isolate` from example Vitest configs
- Streamline Test addon introduction
- Address feedback from EAP

#### Manual testing

1. Sync this branch with locally running docs

### Documentation

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [x] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

## Greptile Summary

Here's my summary of the key documentation changes for the Storybook Test addon:

Updated documentation for Storybook Test addon v8.5, introducing enhanced testing capabilities and improved integration with coverage and accessibility testing.

- Added comprehensive coverage reporting functionality with configurable watermarks and HTML reports at `/coverage/index.html`
- Integrated accessibility testing with Test addon, providing seamless a11y checks during component testing
- Added framework-specific plugin configurations for React, Vue and Svelte projects
- Updated test runner documentation with clearer examples and improved troubleshooting guidance
- Added installation instructions for coverage providers (v8/Istanbul) across package managers

The changes focus on providing clear guidance for setting up and using the enhanced testing capabilities while maintaining consistency across different frameworks and environments.

<sub>💡 (1/5) You can manually trigger the bot by mentioning @greptileai in a comment!</sub>

### CODE REVIEW

**Review by jonniebigodes — APPROVED** [review]
Thanks for taking the time to put together this pull request and make the required adjustments. I left some items for you to look into when you can. Nothing major. 

Also, just as a reminder, we still have some "TK" and missing images and if we don't get to them, let's comment those out.

I'm approving it to unblock you.

File: docs/writing-tests/test-addon.mdx [hunk 1]
```diff
@@ -321,9 +332,40 @@ export default defineWorkspace([
...
+When you run tests in the [Storybook UI](#storybook-ui), the addon runs Vitest in the background and reports the results in the sidebar. 
+
 ## Configuring tests
 
-Most of the configuration for the Vitest plugin's behavior is done in the Vitest configuration and setup files. However, you can also define configuration in your stories themselves, using [tags](../writing-stories/tags.mdx), to control how they are tested.
+The tests run by the addon can be configured in two ways. You can toggle which test types are run, and you can include, exclude, or skip stories from being tested.
+
+### Toggling test types
+
+You can toggle different testing types on or off in the test module. Press the edit icon to enter edit mode, then toggle the test types you want to run. You may only see a checkbox for coverage:
```
[the remark this answers is not in the document]
Comment (kylegach) [reply docs/writing-tests/test-addon.mdx:349]: Those are two different things. Test addon is the overall experience (what we refer to in marketing as Storybook Test). Test module is the part of the UI that is pinned to the bottom of the sidebar.

That said, this comment gave me an idea of how to make that more clear here. 👍

File: docs/writing-tests/test-coverage.mdx [hunk 2]
```diff
@@ -5,19 +5,222 @@ sidebar:
...
+- `cleanOnRerun`
+- `reportOnFailure`
+- `reporter`
+- `reportsDirectory`
+
+### Troubleshooting
+
+#### Excluding stories from the coverage report
+
+Until Vitest 2.2.0 is released, the generated coverage report will include the stories files themselves. This is misleading and they should be excluded. To do this, you can add the following to your Vitest config:
```
Comment (JReinhold) [inline docs/writing-tests/test-coverage.mdx:195]: `3.0.0`, not `2.2.0`, as they scrapped that minor in favor of a major release instead. See https://github.com/vitest-dev/vitest/releases/tag/v3.0.0-beta.1

**Review by JReinhold — COMMENTED** [review]
Looks great!

**Pull request merged**